moviepy/audio/io/readers.py

```diff
@@ -119,6 +119,9 @@ def read_chunk(self,chunksize):
         result = (1.0*result / 2**(8*self.nbytes-1)).\
                                  reshape((int(len(result)/self.nchannels),
                                           self.nchannels))
+        pad = np.zeros((chunksize - len(result), self.nchannels),
+                       dtype=result.dtype)
+        result = np.concatenate([result, pad])
```

Comment:
Pad the read chunk with zeros when there isn't enough audio left to read, so the buffer is always at full length.